### PR TITLE
feat: Convert to context succeeding Part 3

### PR DESCRIPTION
**Pull request by samuel-hawker**
Switched to use context.succeeding(handler)
This means the boilerplate code of handling failures
is no longer needed in many tests and the Future is
passed to the handler already unwrapped

Also used HamCrest matchers in appropriate places to
increase the verbosity and usefulness of failed test results

Also in this PR:
Minor test renames, variable renames and test cleanup

### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by scholzj** [comment]
@samuel-hawker Looks like one of the other PRs created some conflicts. Could you have a look please?

**Comment by scholzj** [comment]
@samuel-hawker Couls you look at the Travis results? It seems to complain abotu this:

```
[ERROR] COMPILATION ERROR : 
[ERROR] /home/travis/build/strimzi/strimzi-kafka-operator/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperatorTest.java:[136,5] method does not override or implement a method from a supertype
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:testCompile (default-testCompile) on project cluster-operator: Compilation failure
[ERROR] /home/travis/build/strimzi/strimzi-kafka-operator/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperatorTest.java:[136,5] method does not override or implement a method from a supertype
```

**Comment by scholzj** [comment]
@strimzi-ci run tests

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaConnectCrdOperatorIT.java [hunk 1]
```diff
@@ -119,86 +120,51 @@ protected KafkaConnect getResource() {
...
-        PlatformFeaturesAvailability.create(vertx, client).setHandler(pfaRes -> {
-            if (pfaRes.succeeded())    {
-                pfa.set(pfaRes.result());
-                versionAsync.countDown();
-            } else {
-                context.failNow(pfaRes.cause());
-            }
-        });
-        if (!versionAsync.await(60, TimeUnit.SECONDS)) {
-            context.failNow(new Throwable("Test timeout"));
```
Comment (scholzj) [inline operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaConnectCrdOperatorIT.java]: I wonder if these timeouts had any special meanings and how / when the test timeouts without them. Have you done any tests around this?

Comment (samuel-hawker) [reply operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaConnectCrdOperatorIT.java]: I am glad you brought this up, I should have documented this better as I raised the PR.

I have switched to using the `Checkpoint` object, this comes with the default vertx test timeout (though can be overridden as you have for several tests in other classes using e.g. (    `@Timeout(value = 2, timeUnit = TimeUnit.MINUTES)`)
I can see for some tests we might need to have to override the default timeout like above, but for the tests i refactored this was not needed.

For each of the steps i wrapped them in a `context.succeeding`, `failing`, or `verify` all of these will immediately call `context.failNow` with all relevant information if their assertions aren't met.

By using a single checkpoint at the end of a test we can gate its completion on reaching the end without throwing an exception (unless the exception is expected ofcourse).

In the event that none of the assertions fail, the checkpoint will still timeout and cause a failNow reporting that it had timed out. In cases like these the log lines are most useful for inferring where we didn't reach up to, but we can easily mitigate this, if it is not good enough, by adding in more checkpoints that are incremental throughout the test.

Comment (scholzj) [reply operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaConnectCrdOperatorIT.java]: Great, thanks for the explanation. 👍

Comment (samuel-hawker) [reply operator-common/src/test/java/io/strimzi/operator/common/operator/resource/KafkaConnectCrdOperatorIT.java]: Anytime!

**Comment by scholzj** [comment]
Thanks for the PR

**Pull request merged**